analysis/tools/InteractiveCLI/src/main/kotlin/de/maibornwolff/codecharta/tools/interactivecli/ParserDialogInterface.kt

```diff
@@ -0,0 +1,18 @@
+package de.maibornwolff.codecharta.tools.interactivecli
+
+import picocli.CommandLine
+
+interface ParserDialogInterface {
+    fun generateParserArgs(args: Array<String>, commandLine: CommandLine): Array<String>
```

Comment:
I'd recommend to use `List` instead of `Array`

Reply:
Fixed in the next commits.
But is there a specific reason? Because I would still have to cast the result back to an array later.

Reply:
https://stackoverflow.com/a/36263748

The main function is derived from java and uses an Array as input. We can't change that. However, we can decide what we use in our code and we should prefer lists there (see link above).
In that case it's fine to simply convert from array to list and list to array when dealing with the main function.

Reply:
Aha got you! thanks for the explanation 👍